src/config.rs

```diff
@@ -28,6 +27,63 @@ pub enum ConfigError {
     AmbiguousSource(PathBuf, PathBuf),
 }
 
+/// Set of configs which can be merged as needed.
+///
+/// Sources from the lowest precedence:
+/// 1. Default
+/// 2. Base environment variables
+/// 3. User config `~/.jjconfig.toml` or `$JJ_CONFIG`
+/// 4. TODO: Repo config `.jj/repo/config.toml`
+/// 5. TODO: Workspace config `.jj/config.toml`
+/// 6. Override environment variables
+/// 7. TODO: Command-line arguments `--config-toml`
+#[derive(Clone, Debug)]
+pub struct LayeredConfigs {
+    default: config::Config,
+    env_base: config::Config,
+    user: Option<config::Config>,
+    env_overrides: config::Config,
+}
+
+impl Default for LayeredConfigs {
```

Comment:
I think it might be better to make instantiation of this struct explicit with something like `LayeredConfigs::from_environment()`. It feels like this does too much to be in `Default::default()`. What do you think?

Reply:
> It feels like this does too much to be in `Default::default()`.

Agreed. I tend to avoid `from_<word>` for non-conversion constructor, but `from_environment()` sounds good. will update, thanks.